From eacc60dbb2eff0d067c1501b009eea058d3ce368 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ernst=20Sj=C3=B6strand?= <ernst@stickybit.se>
Date: Wed, 8 Feb 2017 16:19:31 +0100
Subject: [PATCH 3/3] cross toolchain: Use clang properly

The target name will be dynamically patch from the recipe.
---
 cross/arm/toolchain.cmake | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/cross/arm/toolchain.cmake b/cross/arm/toolchain.cmake
index fe08a7a..fc7b910 100644
--- a/cross/arm/toolchain.cmake
+++ b/cross/arm/toolchain.cmake
@@ -7,14 +7,16 @@ set(CMAKE_SYSTEM_PROCESSOR armv7l)
 ## Specify the toolchain
 set(TOOLCHAIN "arm-linux-gnueabihf")
 
-add_compile_options(-target armv7-linux-gnueabihf)
+add_compile_options(-target ${TOOLCHAIN})
 add_compile_options(-mthumb)
 add_compile_options(-mfpu=vfpv3)
+add_compile_options(-mfloat-abi=hard)
+add_compile_options(-march=armv7l)
 add_compile_options(--sysroot=${CROSS_ROOTFS})
 
 set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} -target ${TOOLCHAIN}")
-set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} -B ${CROSS_ROOTFS}/usr/lib/gcc/${TOOLCHAIN}")
-set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} -L${CROSS_ROOTFS}/lib/${TOOLCHAIN}")
+set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} -fuse-ld=$ENV{GCC_TOOLCHAIN}/${TOOLCHAIN}-ld")
+set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} -mthumb -mfpu=vfpv3 -mfloat-abi=hard -march=armv7l")
 set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} --sysroot=${CROSS_ROOTFS}")
 
 set(CMAKE_EXE_LINKER_FLAGS    "${CMAKE_EXE_LINKER_FLAGS}    ${CROSS_LINK_FLAGS}" CACHE STRING "" FORCE)
-- 
2.9.3

